tools/installer/installer.py

```diff
@@ -0,0 +1,626 @@
+        try:
+            value = func(*args)
+        except subprocess.CalledProcessError as err:
+            if err.returncode and _allow_error:
+                return err
+            failure_exit(err)
+        return value
+
+    return wrapper
+
+
+def default_answer(func):
+    @functools.wraps(func)
+    def wrapper(*args, **kwds):
+        _default = kwds.get('default', "")
+        if _default:
+            args = list(args)
+            args[-1] += f"[{_default}] {os.linesep}"
+        value = func(*args)
+        return value if value != "" else _default
+
+    return wrapper
+
+
+class Installer():
+    def __init__(self, interviewer):
+        self.version = interviewer.release.version
+        self.release = interviewer.release
+        self.verbose = True
+        self.interviewer = interviewer
+
+    @property
+    def binary_path(self):
+        return self.get_binary_path()
+
+    def get_binary_path(self):
+        return os.path.join(os.path.realpath(os.path.curdir), DEFAULT_BINARY_NAME)
+
+    @property
+    def cosmovisor_service_cfg(self):
+        return f"""
+[Unit]
+Description=Service for running cheqd-node daemon
+After=network.target
+Documentation=https://docs.cheqd.io/node
+
+[Service]
+Environment="DAEMON_HOME={self.cheqd_root_dir}"
+Environment="DAEMON_NAME={DEFAULT_BINARY_NAME}"
+Environment="DAEMON_ALLOW_DOWNLOAD_BINARIES=true"
+Environment="DAEMON_RESTART_AFTER_UPGRADE=true"
+Environment="UNSAFE_SKIP_BACKUP=true"
+Type=simple
+User=cheqd
+ExecStart=/usr/bin/cosmovisor run start
+Restart=on-failure
+RestartSec=30
+StartLimitBurst=5
+StartLimitInterval=60
+TimeoutSec=120
+StandardOutput=syslog
+StandardError=syslog
+SyslogFacility=syslog
+SyslogIdentifier=cosmovisor
+LimitNOFILE=65535
+
+[Install]
+WantedBy=multi-user.target
+"""
+
+    @property
+    def default_logrotate_cfg(self):
+        return """
+%s/stdout.log {
+  rotate 7
+  daily
+  maxsize 100M
+  notifempty
+  copytruncate
+  compress
+  maxage 7
+}
+""" % self.cheqd_log_dir
+
+    @property
+    def default_rsyslog_cfg(self):
+        binary_name = "cosmovisor" if self.interviewer.is_cosmo_needed else "cheqd-noded"
+        return f"""
+if $programname == '{binary_name}' then {self.cheqd_log_dir}/stdout.log
+& stop
+"""
+
+    @property
+    def cheqd_root_dir(self):
+        return os.path.join(self.interviewer.home_dir, ".cheqdnode")
+
+    @property
+    def cheqd_config_dir(self):
+        return os.path.join(self.cheqd_root_dir, "config")
+
+    @property
+    def cheqd_data_dir(self):
+        return os.path.join(self.cheqd_root_dir, "data")
+
+    @property
+    def cheqd_log_dir(self):
+        return os.path.join(self.cheqd_root_dir, "log")
+
+    @property
+    def cosmovisor_root_dir(self):
+        return os.path.join(self.cheqd_root_dir, "cosmovisor")
+
+    @property
+    def cosmovisor_cheqd_bin_path(self):
+        return os.path.join(self.cosmovisor_root_dir, f"current/bin/{DEFAULT_BINARY_NAME}")
```

Comment:
Should this be `/usr/bin`?

Reply:
In case using Cosmovisor, bin is in `/home/cheqd/.cheqdnode/cosmovisor/genesis/bin/`. Symlink from `/usr/bin/cheqd-noded` to path above.